contracts/CurrencyFactory.sol

```diff
@@ -151,6 +168,17 @@ contract CurrencyFactory is Standard223Receiver, TokenHolder {
   	return (clnAddress == _token || currencyMap[_token].totalSupply > 0);
   }
 
+  /// @dev sets tokenURI for the given currency, can be used during the sell only
+  /// @param _token address address of the token to update
+  /// @param _tokenURI string the URI may point to a JSON file that conforms to the "Metadata JSON Schema".
+  function setTokenURI(address _token, string _tokenURI) public
+                              tokenIssuerOnly(_token, msg.sender)
+                              marketClosed(_token)
```

Comment:
This means that you force the owner to accept ownership of the token if he wants to change the metadata after the market is opened. I guess it is a good thing...